inginious/frontend/templates/task_dispensers/toc.html

```diff
@@ -4,37 +4,39 @@
 {# Recursively print a list of sections and their content #}
 
 {% set level = level|default(3, true) %}
-
 {% for section in sections %}
-    {% if section.is_terminal() %}
-        {% set ns = namespace(section_tasks=[]) %}
-        {% for taskid in section.get_tasks() if taskid in tasks %}
-            {% set ns.section_tasks = ns.section_tasks + [tasks[taskid]] %}
-        {% endfor %}
-        {% with tasks=ns.section_tasks, tag_list=tag_filter_list, title=section.get_title(), config=section.get_config() %}
-            {% include "task_dispensers/task_list.html" %}
-        {% endwith %}
-    {% else %}
-        <div class="section sections-list mb-4">
-            <div class="section_header d-flex justify-content-between divided pr-3" onclick="dropdown(this)">
-                <span class="title h{{level}} mr-3">
-                    {{ _(section.get_title()) }}
-                </span>
-                <span class="divider"></span>
+    {% set ns = namespace(section_tasks=[],have_accessible_tasks=False) %}
+    {% for taskid in section.get_tasks() if taskid in tasks %}
+        {% set ns.section_tasks = ns.section_tasks + [tasks[taskid]] %}
+        {% set ns.have_accessible_tasks = ns.have_accessible_tasks or user_manager.user_is_superadmin() or accessibilities[user_manager.session_username()][taskid].is_open()%}
```

Comment:
```suggestion
        {% set ns.have_accessible_tasks = ns.have_accessible_tasks or accessibilities[user_manager.session_username()][taskid].is_open()%}
```
It has been decided with the introduction of task dispensers to make the student view visible as a student to the course administrator. Bypassing the check for administrators would confuse the user in parameterizing the course table of contents.

But bypassing for superadmins only is even more confusing and doesn't seem to make sense ?

Reply:
You're right... Forgot the superadmin case...